src/rulesets/Base/arraymath.jl

```diff
@@ -62,13 +73,30 @@ function rrule(
     return A * B, times_pullback
 end
 
+# Not inplace for if Arrays of Arrays are involved
+# TODO: remove this after https://github.com/JuliaLang/julia/issues/38772
+# and make the above accept any MatMulField
+function rrule(
+    ::typeof(*),
+    A::AbstractVector{<:MatMulField},
+    B::AbstractMatrix{<:MatMulField},
+)
+    function times_pullback(Ȳ)
+        @assert size(B, 1) === 1   # otherwise primal would have failed.
+        return NO_FIELDS, @thunk(Ȳ * vec(B')), @thunk(A' * Ȳ)
+    end
+    return A * B, times_pullback
+end
+
+
+
 function rrule(
-   ::typeof(*), A::CommutativeMulNumber, B::AbstractArray{<:CommutativeMulNumber}
+   ::typeof(*), A::CommutativeMulNumber, B::AbstractArray{<:MatMulField}
 )
     function times_pullback(Ȳ)
         return (
             NO_FIELDS,
-            @thunk(dot(Ȳ, B)'),
+            (dot(Ȳ, B)'),
```

Comment:
Swapping the order in `dot` conjugates the output, e.g.
```julia
julia> x, y = randn(ComplexF64, 3), randn(ComplexF64, 3);

julia> dot(x, y) == dot(y, x)'
true
```
Could make the same change in the below rule

```suggestion
            dot(B, Ȳ),
```